bfvmm/include/vmcs/bitmap.h

```diff
@@ -0,0 +1,66 @@
+//
+// Bareflank Hypervisor
+//
+// Copyright (C) 2015 Assured Information Security, Inc.
+// Author: Rian Quinn        <quinnr@ainfosec.com>
+// Author: Brendan Kerrigan  <kerriganb@ainfosec.com>
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#ifndef BITMAP__H
+#define BITMAP__H
+
+#include <stdint.h>
+
+class bitmap
+{
+public:
+    /// Constructor
+    /// @param num_bits size of the bitmap in bits
+    ///
+    bitmap(uint32_t num_bits);
```

Comment:
agreed there is no reason to alloc that much memory, but if that were your reason, you should change it to 16bits as that's more realistic. My main reason for enforcing all types to be 64bit is it's faster on 64bit, and I want to limit how much we are arbitrarily defining type sizes. Basically.... unless you specifically need it to be less than 64 bits (for example, the use of uint8_t * is fine), just use 64bits.